src/services/company.js

```diff
@@ -53,51 +53,55 @@ class CompanyService {
     /**
      * @param {*} company_id Id of the company
      * @param {*} showHidden weather to show the company if it is hidden, defaults to false
-     * @param {*} showAdminReason weahter to show the admin reason given to hide this company, defaults to false
+     * @param {*} showAdminReason weather to show the admin reason given to hide this company, defaults to false
      */
-    findById(company_id, showHidden = false, showAdminReason = false) {
+    async findById(company_id, showHidden = false, showAdminReason = false) {
         const query = Company.findById(company_id);
         if (!showHidden) query.withoutBlocked().withoutDisabled();
         if (!showAdminReason) query.hideAdminReason();
-        return query;
+        const company = await query;
+        return  company;
     }
 
     /**
      * @param {@param} companyId Id of the company
      */
-    block(companyId, adminReason) {
-        return Company.findByIdAndUpdate(
+    async block(companyId, adminReason) {
+        const company =  await Company.findByIdAndUpdate(
             companyId,
             {
                 isBlocked: true,
                 adminReason
             },
             { new: true });
+        return company;
```

Comment:
same question for all the changes of this type

Reply:
This was because of #246, there seems to be a difference between returning a promise or a query